### ISSUE TITLE
TwoStreamControlNet: control_scale is always set to 1.0

### ISSUE THREAD

**Issue by deltheil** [vislearn/ControlNet-XS#17]
It is [initialized](https://github.com/vislearn/ControlNet-XS/blob/0840b7ec0477cab94d6c5367c18f0566becb24e2/sgm/modules/diffusionmodules/twoStreamControl.py#L93) in the CTOR and never changes afterwards.

As a consequence, setting `control_scale` in the high-level `get_sdxl_sample` API has no effect on it while this factor is involved in the [timestep embedding computation](https://github.com/vislearn/ControlNet-XS/blob/0840b7ec0477cab94d6c5367c18f0566becb24e2/sgm/modules/diffusionmodules/twoStreamControl.py#L318-L322
).

(In other words, the base model `time_embed` has no effect whatever the input `control_scale`)

But also, setting `control_scale=0.0` when calling `get_sdxl_sample` is different than completely disabling ControlNet-XS via [`no_control = True`](https://github.com/vislearn/ControlNet-XS/blob/0840b7ec0477cab94d6c5367c18f0566becb24e2/sgm/modules/diffusionmodules/twoStreamControl.py#L92).

It looks to me, it should be changed via something like:

```diff
--- a/scripts/control_utils.py
+++ b/scripts/control_utils.py
@@ -185,6 +185,7 @@ def get_sdxl_sample(

     if float(control_scale) != 1.0:
         model.model.scale_list *= control_scale
+        model.model.control_scale = control_scale
         print(f'[CONTROL CORRECTION OF {type(model).__name__} SCALED WITH {control_scale}]')

     control = torch.stack([tt.ToTensor()(ds['hint'][..., None].repeat(3, 2))] * num_samples).float().to('cuda')
@@ -231,6 +232,7 @@ def get_sdxl_sample(

     #  reset scales
     model.model.scale_list = model.model.scale_list * 0. + 1.
+    model.model.control_scale = 1.

     return x_samples, control
```

--

Is it intended (bug or feature)? Am I missing something?

**Comment by Sipirius** [comment]
This functionality should not have been removed and will be added back shortly.

cheers

**Comment by deltheil** [comment]
> This functionality should not have been removed and will be added back shortly.

Looks like this has been fixed by 3df4d6e
